contracd/src/dbusinterface.h

```diff
@@ -26,7 +26,7 @@ class DBusInterface : public QObject
     Q_PROPERTY(qint32 rssiCorrection READ rssiCorrection WRITE setRssiCorrection NOTIFY rssiCorrectionChanged)
 
 public:
-    explicit DBusInterface(QObject *parent = nullptr);
+    explicit DBusInterface(bool &result, QObject *parent = nullptr);
```

Comment:
Could this be amended so that the parameter is optional (as the `parent` parameter is already)? I'm not sure whether it's possible to do this using a reference, so it may be necessary to switch to using a pointer instead.

Reply:
Yep only default option without copying afaik would be having a `nullptr` instead. But then we have to add some extra complexity so that the results never get written to `nullptr`, haven't we?

Reply:
Yes, I guess one approach would be to retain the local bool to store the result inside the initialiser, and at the end of initialisation have something like this:
```
    if (success) {
        *success = result;
    }
```